### PR TITLE
comple servlets and jsp pages

### PR DESCRIPTION
**Pull request by AndrewLinkii**

### CODE REVIEW

File: src/main/java/shop/controllers/AllProductsController.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import shop.lib.Injector;
+import shop.model.Product;
+import shop.service.ProductService;
+
+public class AllProductsController extends HttpServlet {
```
Comment (dmytro-hryhoruk) [inline src/main/java/shop/controllers/AllProductsController.java:13]:
```suggestion
public class GetAllProductsController extends HttpServlet {
```

File: src/main/java/shop/controllers/AllUsersController.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import shop.lib.Injector;
+import shop.model.User;
+import shop.service.UserService;
+
+public class AllUsersController extends HttpServlet {
```
Comment (dmytro-hryhoruk) [inline src/main/java/shop/controllers/AllUsersController.java:13]:
```suggestion
public class GetAllUsersController extends HttpServlet {
```

File: src/main/java/shop/controllers/RegistrationController.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String name = req.getParameter("user_name");
+        String password = req.getParameter("psw");
+        String pswRe = req.getParameter("psw-repeat");
+
+        if (password.equals(pswRe)) {
+            User user = new User(name, login, password);
+            ShoppingCart shoppingCart = new ShoppingCart(null, user);
```
Comment (dmytro-hryhoruk) [inline src/main/java/shop/controllers/RegistrationController.java:37]: let's just create constructor which will accept only user

File: src/main/java/shop/controllers/ShopCartController.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import javax.servlet.http.HttpServletResponse;
+import shop.lib.Injector;
+import shop.model.Product;
+import shop.model.ShoppingCart;
+import shop.service.ShoppingCartService;
+
+public class ShopCartController extends HttpServlet {
+    private static final Long USER_ID = 1L;
+
+    private static final Injector injector = Injector.getInstance("shop");
```
Comment (dmytro-hryhoruk) [inline src/main/java/shop/controllers/ShopCartController.java:17]:
```suggestion
    private static final Injector INJECTOR = Injector.getInstance("shop");
```

File: src/main/webapp/WEB-INF/views/addProduct.jsp [hunk 5]
```diff
@@ -0,0 +1,23 @@
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/WEB-INF/views/addProduct.jsp:1]: remove empty lines

File: src/main/java/shop/controllers/RegistrationController.java (hunk 3 above)
Comment (AndrewLinkii) [reply src/main/java/shop/controllers/RegistrationController.java:37]: Maybe it's better to create just an empty ArrayList?

Comment (dmytro-hryhoruk) [reply src/main/java/shop/controllers/RegistrationController.java]: you can initialized your list in your constructor but it doesn't have to accept it as parameter

File: src/main/java/shop/controllers/AddProductToShoppingCart.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    private final ShoppingCartService shoppingCartService =
+            (ShoppingCartService) INJECTOR.getInstance(ShoppingCartService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Long productId = Long.valueOf(req.getParameter("id"));
+        Product product = productService.get(productId);
+        ShoppingCart shoppingCart = shoppingCartService.getByUserId(USER_ID);
+        shoppingCartService.addProduct(shoppingCart,product);
```
Comment (dmytro-hryhoruk) [inline src/main/java/shop/controllers/AddProductToShoppingCart.java:28]:
```suggestion
        shoppingCartService.addProduct(shoppingCart, product);
```

File: src/main/java/shop/controllers/ShopCartController.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import shop.lib.Injector;
+import shop.model.Product;
+import shop.model.ShoppingCart;
+import shop.service.ShoppingCartService;
+
+public class ShopCartController extends HttpServlet {
```
Comment (dmytro-hryhoruk) [inline src/main/java/shop/controllers/ShopCartController.java:14]:
```suggestion
public class GetShoppingCartController extends HttpServlet {
```

File: src/main/webapp/web.xml [hunk 8]
```diff
@@ -5,12 +5,93 @@
...
+
+    <servlet>
+        <servlet-name>addProductToShopCart</servlet-name>
+        <servlet-class>shop.controllers.AddProductToShoppingCart</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>shoppingCart</servlet-name>
+        <url-pattern>/shoppingCart</url-pattern>
+    </servlet-mapping>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml:90]: it doesn't matter for program but it's just better to read first <servlet> and than it's mapping

**Pull request merged**